khook/engine.c

```diff
@@ -116,13 +120,16 @@ int khook_init(void)
 	// have eXecutable attributes. That's why we have to mark the
 	// region executable explicitly.
 	//
-
 	set_memory_x = (void *)khook_lookup_name("set_memory_x");
-	if (set_memory_x) {
-		int numpages = round_up(KHOOK_STUB_TBL_SIZE, PAGE_SIZE) / PAGE_SIZE;
+#ifdef  USE_CR0_FAM
+	if (set_memory_x)
 		set_memory_x((unsigned long)khook_stub_tbl, numpages);
-	}
-
+#endif
+#ifdef  USE_PTE_FAM
+	do {	/* Yes, I pass <= 100 symbol in one line limitation :) */
+		set_addr_ex((unsigned long)khook_stub_tbl + ((-1 + numpages--) * PAGE_SIZE) + 0x01);
```

Comment:
`set_memory_x` should perform the same work, no?

Reply:
Yeah, but, this is alternative method.